docs/data/joy/components/accordion/AccordionTransition.js

```diff
@@ -0,0 +1,86 @@
+import * as React from 'react';
+import Box from '@mui/joy/Box';
+import AccordionGroup from '@mui/joy/AccordionGroup';
+import Accordion from '@mui/joy/Accordion';
+import AccordionDetails from '@mui/joy/AccordionDetails';
+import AccordionSummary from '@mui/joy/AccordionSummary';
+import RadioGroup from '@mui/joy/RadioGroup';
+import Radio from '@mui/joy/Radio';
+import Stack from '@mui/joy/Stack';
+import Typography from '@mui/joy/Typography';
+import HighlightedCode from 'docs/src/modules/components/HighlightedCode';
+import BrandingProvider from 'docs/src/BrandingProvider';
+
+export default function AccordionTransition() {
+  const [transition, setTransition] = React.useState('0.4s ease');
+  return (
+    <Stack alignItems="center" spacing={2} sx={{ flex: 1 }}>
+      <RadioGroup
+        orientation="horizontal"
+        value={transition}
+        onChange={(event) => {
+          setTransition(event.target.value);
+        }}
+      >
+        <Radio value="0.3s linear" label="Linear" />
+        <Radio value="0.4s ease" label="Easing" />
+        <Radio value="0.8s cubic-bezier(0.68, -0.6, 0.32, 1.6)" label="Bouncy" />
+        <Radio value="mix" label="Mix" />
+      </RadioGroup>
+      <AccordionGroup
+        transition={
```

Comment:
Would it be clearer if this was customized with CSS and the sx prop? Developer would have direct access to their CSS animation design tokens, like duration, easing, etc.

What problem does the `transition` prop solve better?

Reply:
I thought about it as well. With `transition` prop, developers don't have to remember and specify both properties `grid-template-rows` and `padding`.

Also, if you want to customize the transition for both initial and expanded state in CSS, you will need to do this:

```js
<AccordionGroup
  sx={{
    '--AccordionDetails-transition': `grid-template-rows 0.2s, padding-block 0.2s`,
        [`& .${accordionDetailsClasses.root}.${accordionDetailsClasses.expanded}`]: {
          '--AccordionDetails-transition': `grid-template-rows 0.3s, padding-block 0.3s`,
        },
  }}
/>
```

whereas with transition prop:
```js
<AccordionGroup
  transition={{
    initial: '0.2s',
    expanded: '0.3s',
  }}
/>
```